src/app/Common/Utils/ServiceCollectionExtensions.cs

```diff
@@ -52,13 +54,24 @@ public static class ServiceCollectionExtensions
         return services;
     }
 
-    public static void ConfigureHealthChecks(this IServiceCollection services, DbSettings dbSettings)
+    public static void ConfigureHealthChecks(this IServiceCollection services, DbSettings dbSettings, CacheSettings cacheSettings)
     {
-        services
-            .AddHealthChecks()
-            .AddMongoDb(
-                mongodbConnectionString: dbSettings.ConnectionString,
-                mongoDatabaseName: dbSettings.Database
-            );
+        var builder = services.AddHealthChecks();
+
+        builder.AddMongoDb(
+            mongodbConnectionString: dbSettings.ConnectionString,
+            mongoDatabaseName: dbSettings.Database);
+
+        if (cacheSettings.IsEnabled)
+        {
+            builder.AddRedis(cacheSettings.ConnectionString);
+        }
+    }
+
+    public static void ConfigureCache(this IServiceCollection services, CacheSettings cacheSettings)
+    {
+        services.AddStackExchangeRedisCache(options => options.Configuration = cacheSettings.ConnectionString);
+
+        services.AddScoped<ICache, Cache>();
```

Comment:
I guess, there're no particular reasons to keep a single instance of ICache per request, so we can make it transient (this is discussable).